File: Single-Linked-List/InsertAtBeginning.java
import java.util.Scanner;

// Node class
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

// Singly Linked List class
public class SinglyLinkedList {

    Node head = null;

    // Method to insert at beginning
    void insertBeg() {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter data for new node: ");
        int value = sc.nextInt();

        Node newNode = new Node(value);

        newNode.next = head;   // Point to current head
        head = newNode;        // Update head
    }
}

File: Single-Linked-List/insert_at_beginning.py
# Node class
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

# Singly Linked List class
class SinglyLinkedList:
    def __init__(self):
        self.head = None

    # Method to insert at beginning
    def insert_beg(self):
        value = int(input("Enter data for new node: "))
        new_node = Node(value)

        new_node.next = self.head   # Link new node
        self.head = new_node        # Update head
